### PR TITLE
[Polkadot Wiki Migration] - Consensus Protocols

### PR DESCRIPTION
**Pull request by 0xLucca**
This page was migrated from https://wiki.polkadot.network/docs/learn-consensus

### CODE REVIEW

**Review by CrackTheCode016 — APPROVED** [review]
This page reads well to me as-is, it could be cool in the future to link a relevant doc / guide for what it would take to implement your own consensus, but that's further down the line :)

File: polkadot-protocol/protocol-operations/consensus-protocols.md
Comment (nhussein11) [inline polkadot-protocol/protocol-operations/consensus-protocols.md:51]:
```suggestion
When multiple validators are block producer candidates in a given slot, all will produce and broadcast a block to the network. At that point, it becomes a race to reach most of the network first. Depending on network topology and latency, both chains will continue building in some capacity until finalization. The [Fork Choice](#fork-choice) rule determines which chain to follow.
```

File: polkadot-protocol/protocol-operations/consensus-protocols.md
Comment (nhussein11) [inline polkadot-protocol/protocol-operations/consensus-protocols.md:74]: Grammarly suggestion

```suggestion
A pure Nakamoto consensus blockchain that runs PoW can only achieve probabilistic finality and eventual consensus. Probabilistic finality means that under some assumptions about the network and participants, we can estimate the probability that it is final if we see a few blocks building on a given block. Eventual consensus means that at some point in the future, all nodes will agree on the truthfulness of one set of data. This eventual consensus may take a long time, and will not be able to determine how long it will take ahead of time. However, finality gadgets such as GRANDPA (GHOST-based Recursive ANcestor Deriving Prefix Agreement) or Ethereum's Casper FFG (the Friendly Finality Gadget) are designed to give more robust and quicker guarantees on the finality of blocks - specifically, that they can never be reverted after some process of Byzantine agreements has taken place. The notion of irreversible consensus is known as provable finality.
```

**Review by nhussein11 — APPROVED** [review]
LGTM!

**Comment by dawnkelly09** [comment]
@eshaben I copied this file, renamed it, and moved it into its home in the updated IA. The original page is still in the legacy folder with a TODO for me to delete once this page is approved.  There is one image on this page. I updated the /image subdirs to match new IA for this page, moved image, and updated the path in the snippet. Needs a review from you now. Thanks!

**Review by eshaben — APPROVED** [review]
LGTM

**Pull request merged**
